### PR TITLE
feature: S3C-2002 introduce metadata service policy

### PR DESCRIPTION
**Pull request by rahulreddy**
This allows creating and assigning policies to users to access
metadata proxy over cloudsever.

### CODE REVIEW

**Review by anurag4DSB — APPROVED** [review]

**Comment by rahulreddy** [comment]
@bert-e reset

**Review by JianqinWang — APPROVED** [review]

File: lib/policyEvaluator/RequestContext.js [hunk 1]
```diff
@@ -174,6 +181,15 @@ function _buildArn(service, generalResource, specificResource, requesterInfo) {
         }
         return `arn:scality:sso:::${generalResource}`;
     }
+    if (service === 'metadata') {
+        // arn:aws:metadata::<account-id>:<resource-type><resource>
```
Comment (alexanderchan-scality) [inline lib/policyEvaluator/RequestContext.js]: minor:
```suggestion
        // arn:scality:metadata::<account-id>:<resource-type><resource>
```

**Comment by rahulreddy** [comment]
@bert-e reset

**Comment by rahulreddy** [comment]
@bert-e reset

File: lib/policyEvaluator/RequestContext.js [hunk 2]
```diff
@@ -123,11 +123,18 @@ function _findAction(service, method) {
     if (service === 'sts') {
         return _actionMapSTS[method];
     }
+    if (service === 'metadata') {
+        const actions = {
```
Comment (jonathan-gramain) [inline lib/policyEvaluator/RequestContext.js]: minor: for consistency why not having a top-level `_actionMapMetadata` object instead

**Review by jonathan-gramain — APPROVED** [review]

File: lib/policyEvaluator/RequestContext.js (hunk 2 above)
Comment (rahulreddy) [reply lib/policyEvaluator/RequestContext.js]: No reason. Originally I intended to have one action mapped to all metadata requests but decided later to split up admin and bucketd routes. I can do it.

**Comment by rahulreddy** [comment]
@bert-e reset

**Comment by rahulreddy** [comment]
@bert-e force_reset

**Review by jonathan-gramain — APPROVED** [review]

File: lib/policyEvaluator/RequestContext.js [hunk 3]
```diff
@@ -174,6 +182,15 @@ function _buildArn(service, generalResource, specificResource, requesterInfo) {
         }
         return `arn:scality:sso:::${generalResource}`;
     }
+    if (service === 'metadata') {
+        // arn:scality:metadata::<account-id>:<resource-type><resource>
```
Comment (bennettbuchanan) [inline lib/policyEvaluator/RequestContext.js]: `<resource-type><resource>` => `<resource-type>/<resource>`

**Review by bennettbuchanan — APPROVED** [review]

**Comment by rahulreddy** [comment]
@bert-e reset

**Review by bennettbuchanan — APPROVED** [review]

**Comment by rahulreddy** [comment]
@bert-e approve

**Pull request merged**